src/d3a/d3a_core/sim_results/area_statistics.py

```diff
@@ -186,40 +199,52 @@ def _accumulate_house_trades(house, grid, accumulated_trades, past_market_types)
         }
     house_IAA_name = make_iaa_name(house)
     child_names = [c.name for c in house.children]
-    for market in getattr(house, past_market_types):
-        for trade in market.trades:
-            if area_name_from_area_or_iaa_name(trade.seller) in child_names and \
-                    area_name_from_area_or_iaa_name(trade.buyer) in child_names:
-                # House self-consumption trade
-                accumulated_trades[house.name]["produced"] -= trade.offer.energy
-                accumulated_trades[house.name]["earned"] += trade.offer.price
-                accumulated_trades[house.name]["consumedFrom"] = add_or_create_key(
-                    accumulated_trades[house.name]["consumedFrom"], house.name, trade.offer.energy)
-                accumulated_trades[house.name]["spentTo"] = add_or_create_key(
-                    accumulated_trades[house.name]["spentTo"], house.name, trade.offer.price)
-            elif trade.buyer == house_IAA_name:
-                accumulated_trades[house.name]["earned"] += trade.offer.price
-                accumulated_trades[house.name]["produced"] -= trade.offer.energy
-
-    for market in getattr(grid, past_market_types):
-        for trade in market.trades:
-            if trade.buyer == house_IAA_name and trade.buyer != trade.offer.seller:
-                seller_id = area_name_from_area_or_iaa_name(trade.seller)
-                accumulated_trades[house.name]["consumedFrom"] = add_or_create_key(
-                    accumulated_trades[house.name]["consumedFrom"], seller_id, trade.offer.energy)
-                accumulated_trades[house.name]["spentTo"] = add_or_create_key(
-                    accumulated_trades[house.name]["spentTo"], seller_id, trade.offer.price)
-
-    for market in getattr(house, past_market_types):
-        for trade in market.trades:
-            if area_name_from_area_or_iaa_name(trade.seller) == house.name and \
-                    trade.buyer not in child_names:
-                accumulated_trades[house.name]["consumedFromExternal"] += trade.offer.energy
-                accumulated_trades[house.name]["spentToExternal"] += trade.offer.price
-            elif area_name_from_area_or_iaa_name(trade.buyer) == house.name and \
-                    trade.seller not in child_names:
-                accumulated_trades[house.name]["producedForExternal"] -= trade.offer.energy
-                accumulated_trades[house.name]["earnedFromExternal"] += trade.offer.price
+    house_markets = getattr(house, past_market_types)
+    grid_markets = getattr(grid, past_market_types)
```

Comment:
Another question that just came to mind, by grid_markets I assume we are talking about the parent market of the Area, right ? Because if it is really the grid, then this code is only working for 2-level hierarchies, like default_2a, and will not work as soon as Streets are included. I am confident that it is referring to the parent market, but since the naming is misleading here I would like to check it with you too.

Reply:
I tested with default_5 by adding one additional hierarchy and it works for more than two hierarchy too. In order to avoid confusion for future renaming 'grid' -> 'parent'